pkg/virtctl/vm/expand_test.go

```diff
@@ -0,0 +1,170 @@
+	var file *os.File
+	var vmInterface *kubecli.MockVirtualMachineInterface
+	var expandSpecInterface *kubecli.MockExpandSpecInterface
+	var ctrl *gomock.Controller
+	vmName := "testvm"
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+		expandSpecInterface = kubecli.NewMockExpandSpecInterface(ctrl)
+		vmInterface = kubecli.NewMockVirtualMachineInterface(ctrl)
+		vm = kubecli.NewMinimalVM(vmName)
+	})
+
+	It("should fail with missing input parameters", func() {
+		cmd := clientcmd.NewRepeatableVirtctlCommand("expand")
+		Expect(cmd()).NotTo(Succeed())
+	})
+
+	It("should succeed when called on vm", func() {
+		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
+		vmInterface.EXPECT().GetWithExpandedSpec(context.Background(), vmName)
+
+		cmd := clientcmd.NewVirtctlCommand("expand", "--vm", vmName)
+		Expect(cmd.Execute()).To(Succeed())
+	})
+
+	DescribeTable("should succeed when called with ", func(formatName string) {
+		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
+		vmInterface.EXPECT().GetWithExpandedSpec(context.Background(), vmName)
+
+		cmd := clientcmd.NewVirtctlCommand("expand", outputFormat, formatName, "--vm", vmName)
+		Expect(cmd.Execute()).To(Succeed())
+	},
+		Entry("supported format json", "json"),
+		Entry("supported format yaml", "yaml"),
+	)
+
+	It("should fail when called on non existing vm", func() {
+		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
+		vmInterface.EXPECT().GetWithExpandedSpec(context.Background(), nonExistingVM).Return(nil, fmt.Errorf("\"%s\" not found", nonExistingVM))
+
+		cmd := clientcmd.NewRepeatableVirtctlCommand("expand", "--vm", nonExistingVM)
+		err := cmd()
+
+		Expect(err).To(HaveOccurred())
+		Expect(err).Should(MatchError("error expanding VirtualMachine - non-existing-vm in namespace - default: \"non-existing-vm\" not found"))
+	})
+
+	It("should fail when called with non supported output format", func() {
+		cmd := clientcmd.NewRepeatableVirtctlCommand("expand", outputFormat, invalidFormat, "--vm", vmName)
+		err := cmd()
+
+		Expect(err).To(HaveOccurred())
+		Expect(err).Should(MatchError("error not supported output format defined: test-format"))
+	})
+
+	Context("with file input", func() {
+		const (
+			vmSpec = `apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: testvm
+spec:
+  runStrategy: Always
+  template:
+    spec:
+      domain:
+        devices: {}
+        machine:
+          type: q35
+        resources: {}
+        volumes:
+status:
+`
+			invalidYaml = `apiVersion: kubevirt.io/v1kind: VirtualMachine`
+		)
+		BeforeEach(func() {
+			file, err = os.CreateTemp("", "file-*")
```

Comment:
Can we use here `TempDir` from ginkgo that it should automatically cleaned up at the end of the test